### PR TITLE
[RPMs] move ovs scripts into openshift-sdn-ovs

### PR DESCRIPTION
**Pull request by sdodson**
*This change does these things*
* Drop openshift-sdn-simple-setup-node.sh, it's not used
* Creates `openshift-sdn-ovs` subpackage
 * Pulls in openshift-sdn-simple-setup-node.sh from godep
 * Pulls in openshift-ovs-subnet from godep
 * Add openvswitch dependency on `openshift-sdn-ovs` subpackage
 * Add openvswitch service dependency to openshift-node to ensure it's started before the node
* Bump github.com/openshift/openshift-sdn to stop clobbering /etc/sysconfig/docker
* Changes openshift-node from Requires to Wants on docker because there's no longer an intermediate service between docker and openshift-node to provision the bridge

Ultimately this frees us from building and shipping components across two repos as everything from openshift-sdn is already godep'd into origin.

*Known issues*
* On reboot after having enabled `redhat/openshift-ovs-subnet` docker will fail to start via the services, however once openshift-node starts it will initialize the bridge and then restart docker
* If you have previously enabled `redhat/openshift-ovs-subnet` but disable it afterwards you must manually remove DOCKER_NETWORK_OPTIONS from /etc/sysconfig/docker-network as docker will fail without the sdn bridge

*How to enable integrated `openshift-sdn-ovs`*
* Ensure that all previous openshift-sdn packages are removed from master and nodes
* Set `networkPluginName: "redhat/openshift-ovs-subnet"` in your master's config, restart openshift-master
* Install `openshift-sdn-ovs` on all nodes
* Set `networkPluginName: "redhat/openshift-ovs-subnet"` in each node's config, restart openshift-node

### CODE REVIEW

**Comment by sdodson** [comment]
@rajatchopra Please take a look, I think it's ready for merge.

**Comment by rajatchopra** [comment]
Looks good to me.

**Comment by sdodson** [comment]
@brenton mind giving this a merge if you have time to take a look? It's a slight refactoring of the packaging related to sdn plugins.

**Comment by sdodson** [comment]
Hmm, we need a good way to ensure that openvswitch is started too.

**Comment by rajatchopra** [comment]
Should we do this check and start the service from openshift-sdn scripts?

**Comment by sdodson** [comment]
@sosiouxme thoughts on going ahead with merging this in while we sort out https://github.com/openshift/openshift-sdn/issues/73 separately?

**Comment by sdodson** [comment]
Nevermind, if this is fixed by updating openshift-sdn-kube-subnet-setup.sh we may as well wait for that.

File: rel-eng/openshift-node.service [hunk 1]
```diff
@@ -4,7 +4,7 @@ After=docker.service
 After=openshift-master.service
 After=openshift-sdn-master.service
 After=openshift-sdn-node.service
-Requires=docker.service
+Wants=docker.service
```
Comment (sosiouxme) [inline rel-eng/openshift-node.service]: Why this change? Is it to enable to run with non-systemd docker? Or a mistake...

Comment (sdodson) [inline rel-eng/openshift-node.service]: Sorry if this is a repost, the threading looks weird in github so I suspect I screwed something up when replying the first time.

Currently, If you configure a node to use SDN it will permanently configure docker to use lbr0. The problem is that if you were to reboot lbr0 doesn't exist until openshift-node initializes it, so docker would fail, and openshift-node would never start and initialized the bridge. This allows docker to fail, openshift-node starts and the first thing it does is initialize the bridge and restart docker. It's still slightly messy, but thus far I've not seen it fail.

Long term I think we should fix this by reconfiguring the bridge only when bringing up the SDN. To do that we need to

    Have openshift-node write out an environment file to /run/openshift-sdn-ovs-docker-network (or some other ephemeral file that's removed on reboot) with DOCKER_NETWORK_OPTIONS
    Have openshift-sdn-ovs inject that as an optional EnvironmentFile via /usr/lib/systemd/system/docker.service.d/openshift-sdn-ovs.conf

A few examples of this pattern https://github.com/coreos/flannel/blob/master/dist/mk-docker-opts.sh
http://post-office.corp.redhat.com/archives/appinfra-list/2015-April/msg00688.html

**Comment by rajatchopra** [comment]
This looks good to merge.

**Comment by smarterclayton** [comment]
[merge]
